packages/story-editor/src/app/story/useStoryReducer/reducer.ts

```diff
@@ -17,137 +17,141 @@
     }
 
-    case types.ADD_ELEMENTS: {
+    case ActionTypes.AddElements: {
       return reducers.addElements(state, payload);
     }
 
-    case types.DELETE_ELEMENTS: {
+    case ActionTypes.DeleteElements: {
       return reducers.deleteElements(state, payload);
     }
 
-    case types.UPDATE_ELEMENTS: {
+    case ActionTypes.UpdateElements: {
       return reducers.updateElements(state, payload);
     }
 
-    case types.UPDATE_ELEMENTS_BY_RESOURCE_ID: {
+    case ActionTypes.UpdateElementsByResourceId: {
       return reducers.updateElementsByResourceId(state, payload);
     }
 
-    case types.DELETE_ELEMENTS_BY_RESOURCE_ID: {
+    case ActionTypes.DeleteElementsByResourceId: {
       return reducers.deleteElementsByResourceId(state, payload);
     }
 
-    case types.COMBINE_ELEMENTS: {
+    case ActionTypes.CombineElements: {
       return reducers.combineElements(state, payload);
     }
 
-    case types.SET_BACKGROUND_ELEMENT: {
+    case ActionTypes.SetBackgroundElement: {
       return reducers.setBackgroundElement(state, payload);
     }
 
-    case types.ARRANGE_ELEMENT: {
+    case ActionTypes.ArrangeElement: {
       return reducers.arrangeElement(state, payload);
     }
 
-    case types.ARRANGE_GROUP: {
+    case ActionTypes.ArrangeGroup: {
       return reducers.arrangeGroup(state, payload);
     }
 
-    case types.SET_SELECTED_ELEMENTS: {
+    case ActionTypes.SetSelectedElements: {
       return reducers.setSelectedElements(state, payload);
     }
 
-    case types.SELECT_ELEMENT: {
+    case ActionTypes.SelectElement: {
       return reducers.selectElement(state, payload);
     }
 
-    case types.UNSELECT_ELEMENT: {
+    case ActionTypes.UnselectElement: {
       return reducers.unselectElement(state, payload);
     }
 
-    case types.TOGGLE_ELEMENT_IN_SELECTION: {
+    case ActionTypes.ToggleElementInSelection: {
       return reducers.toggleElement(state, payload);
     }
 
-    case types.TOGGLE_LAYER: {
+    case ActionTypes.ToggleLayer: {
       return reducers.toggleLayer(state, payload);
     }
 
-    case types.DUPLICATE_ELEMENTS_BY_ID: {
+    case ActionTypes.DuplicateElementsById: {
       return reducers.duplicateElementsById(state, payload);
     }
 
-    case types.UPDATE_STORY: {
+    case ActionTypes.UpdateStory: {
       return reducers.updateStory(state, payload);
     }
 
-    case types.UPDATE_ANIMATION_STATE: {
+    case ActionTypes.UpdateAnimationState: {
       return reducers.updateAnimationState(state, payload);
     }
 
-    case types.ADD_ANIMATIONS: {
+    case ActionTypes.AddAnimations: {
       return reducers.addAnimations(state, payload);
     }
 
-    case types.COPY_SELECTED_ELEMENT: {
-      return reducers.copySelectedElement(state, payload);
+    case ActionTypes.CopySelectedElement: {
+      return reducers.copySelectedElement(state);
     }
 
-    case types.RESTORE: {
+    case ActionTypes.Restore: {
       return reducers.restore(state, payload);
```

Comment:
@barklund I get the `This expression is not callable. Type 'never' has no call signatures.` error here, any thoughts on why it's happening?

Reply:
The `restore` functions is not properly typed. The return type is inferred (should be `ReducerState` I guess), and it should conform to the proper overload for immer's produce (on that returns the new state rather than modifies the existing draft which we normally use). Maybe this function could even _not_ use immer, as it doesn't actually mutate anything anyway.